godot-ffi/src/lib.rs

```diff
@@ -237,56 +286,74 @@ fn print_preamble(version: GDExtensionGodotVersion) {
 
 /// # Safety
 ///
-/// The interface must have been initialised with [`initialize`] before calling this function.
+/// - Must be called from the main thread.
+/// - The interface must have been initialised with [`initialize`] before calling this function.
+/// - Must only be called once for each `api_level`.
 #[inline]
-pub unsafe fn load_class_method_table(api_level: ClassApiLevel) {
+pub unsafe fn load_class_method_table(api_level: InitLevel) {
     out!("Load class method table for level '{:?}'...", api_level);
     let begin = std::time::Instant::now();
 
     #[cfg(not(feature = "codegen-lazy-fptrs"))]
-    let mut string_names = StringCache::new(get_interface(), builtin_lifecycle_api());
+    // SAFETY: The interface has been initialized.
+    let interface = unsafe { get_interface() };
+
+    #[cfg(not(feature = "codegen-lazy-fptrs"))]
+    // SAFETY: The interface has been initialized.
+    let mut string_names = StringCache::new(interface, unsafe { builtin_lifecycle_api() });
```

Comment:
Maybe to add to the previous comment -- before, I was considering to encapsulate this later in a conditional type, something along
```rs
#[cfg(not(feature = "codegen-lazy-fptrs"))]
struct MethodTableArgs<'a> {
    interface: &'a GDExtensionInterface,
    string_names: StringCache<'a>,
}

#[cfg(feature = "codegen-lazy-fptrs")]
type MethodTableArgs = ();
```

That way, the `args` could be passed always the same way. However, this would need table codegen changes and might not _that_ much simpler, either :thinking: either way, we can leave this for later.

Reply:
i think we can leave it for later, the refactor i mentioned earlier might also change how this would be done.